accounts-db/src/accounts_db.rs

```diff
@@ -2840,6 +2841,9 @@ impl AccountsDb {
                         // all pubkeys in this store also cannot be removed from all stores they are in
                         let affected_pubkeys = &store_count.1;
                         for key in affected_pubkeys {
+                            let purges_bin_index =
+                                self.accounts_index.bin_calculator.bin_from_pubkey(key);
+                            let bin = purges[purges_bin_index].read().unwrap();
```

Comment:
so this is where I was thinking that by the time we got to here we no longer needed to worry about parallel algorithms. So, before getting to `calc_delete_dependencies`, we could get rid of the RwLock on the bins. However, it may be that since we now have RwLock, we can do parallelism here that we previously didn't have. I imagine a read lock with no contention for write locks (which we have here) is irrelevantly fast. @brooksprumo ?

Reply:
Write is an exclusive access, so read will block until write is finished. In current implementation of clean_accounts it appears at the time of calling calc_delete_dependencies there shouldn't be any write locks.

We can track the bin index, but is the additional logic worth it, if the read lock isn't more expensive?

Reply:
> Ah, I see at the top we grab a read lock, not a write lock. Maybe that changes things? The part about panicking still worries me.

We can use `try_read` instead for shared read access. That should work since we don't expect any exclusive access being held by any write() calls at this time.

Reply:
>             let bin = purges[i].read().unwrap();
>             xs.push(bin.as_ref()); // <-- or whatever .as_ref() alternative in order to get a &HashMap

I think this might not work, because RwLock is designed to be released leaving a scope.

Reply:
Otherwise we might build borrowed_purges with RwLockReadGuargd elements before the outer loop starts and use borrowed_purges in both outer and inner loops without having to acquire any more locks.

Reply:
> Or lock all the bins at the top of the function.

This is the same as _before outter loop starts_